### PR TITLE
feat: bEAC-411 - Virtual: Header & Room Capacity Alert

### PR DESCRIPTION
**Pull request by alucio-echen**
See: https://github.com/alucioinc/eeb/pull/138 for more information

### CODE REVIEW

File: alucio/components/Virtual/MeetingControls/MeetingControls.tsx [hunk 1]
```diff
@@ -217,36 +174,7 @@ function MeetingControls(props: HeadBarProps) {
...
-        }
-        break;
-
-      case PresentationMenuOptions[2].id: // -> Sharing
-        if (!isMenuActive(PresentationMenuOptions, 'my_content')) {
-          shareScreen && shareScreen();
-          onStopPresenting && onStopPresenting();
-        }
-        else { // Do this without confirmation, peer product decision
-          shareScreen && shareScreen();
```
Comment (chibchombiano26) [inline alucio/components/Virtual/MeetingControls/MeetingControls.tsx:237]: Probably a TODO will be required?

File: alucio/components/Virtual/Attendee/Attendee.tsx [hunk 2]
```diff
@@ -136,16 +141,27 @@ function Attendee(props: AttendeeProps) {
...
-        onPress={onAccept}
-        style={[styles.button, styles.accept]}>
-        Admit
-      </Button.Kitten>
-      <Button.Kitten
-        onPress={onDeny}
-        style={[styles.button, styles.deny]}>
-        Remove
-      </Button.Kitten>
+      { props.isRoomFull
```
Comment (chibchombiano26) [inline alucio/components/Virtual/Attendee/Attendee.tsx:144]: What about using iffy here?

File: alucio/components/Virtual/MeetingControls/MeetingControls.tsx [hunk 3]
```diff
@@ -1,213 +1,170 @@
...
+  attendeeCount: number,
+  hostName: string,
+  maxAttendees: number,
+  meetingURL: string,
+}
+
+function HostMeetingInfo(props: HostMeetingInfoProps) {
+  const { attendeeCount, hostName, maxAttendees, meetingURL } = props;
+  const [infoBoxToggle, setInfoBoxToggle] = useState(false);
+  const copyLink = () => {
```
Comment (chibchombiano26) [inline alucio/components/Virtual/MeetingControls/MeetingControls.tsx:77]: i know this part is only web, but would be good have a platform === 'web'

File: alucio/components/Virtual/MeetingControls/MeetingControls.tsx (hunk 1 above)
Comment (chibchombiano26) [inline alucio/components/Virtual/MeetingControls/MeetingControls.tsx:237]: it is always confuse have commented code

File: alucio/components/Virtual/Menu/Menu.tsx [hunk 4]
```diff
@@ -116,46 +91,26 @@ interface SideBarOptionProps {
...
-            [styles.optionSelectedText, props.menuOption.active])
-        }>
-        {props.menuOption.active && props.menuOption.activeTitle
-          ? props.menuOption.activeTitle
-          : props.menuOption.title}
-      </Text>
-    </View>
-  )
-
   return (
```
Comment (chibchombiano26) [inline alucio/components/Virtual/Menu/Menu.tsx:120]: not sure if this is required, as far i remember iffy is able to remember a null if the element is not present, not 100% percent sure

File: alucio/components/VirtualModal/VirtualModal.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+interface VirtualModalProps {
+  headerText: string,
+  showCloseButton?: boolean,
+  children: React.ReactNode,
+  onClose?: () => void,
+  width: number,
+}
+
+export default function VirtualModal(props: VirtualModalProps) {
```
Comment (chibchombiano26) [inline alucio/components/VirtualModal/VirtualModal.tsx:42]: this component looks pretty generic, probably a TODO? in ordre to move into lux-ui?

**Comment by chibchombiano26** [comment]
do you mind to add a gif for this one?

**Comment by chibchombiano26** [comment]
image

when the icon is disabled looks weird

**Comment by chibchombiano26** [comment]
here my tests:

https://www.loom.com/share/447923ba79214a1580edd941974ef3f5

File: alucio/components/Virtual/MeetingControls/MeetingControls.tsx [hunk 6]
```diff
@@ -1,89 +1,133 @@
...
+    navigator.clipboard.writeText(meetingURL)
```
Comment (chibchombiano26) [inline alucio/components/Virtual/MeetingControls/MeetingControls.tsx:78]: what about validating the platform here?, i know this is part is web but doesn't hurt have the validation

File: alucio/components/Virtual/MeetingControls/MeetingControls.tsx [hunk 7]
```diff
@@ -276,11 +296,82 @@ function MeetingControls(props: HeadBarProps) {
...
+        }
+        break;
+      case PresentationMenuOptionsEnum.NOTES:
+        props.toggleHostNotes && props.toggleHostNotes();
+        break;
+    }
+  }
+  const PresentationMenuOptions: MenuOption[] = [];
+  if (mode === MODE_TYPE.GUEST) {
+    // PresentationMenuOptions.push(
```
Comment (chibchombiano26) [inline alucio/components/Virtual/MeetingControls/MeetingControls.tsx:331]: do you mind to include a TODO, here it is always weird to see a code commented without any reason

File: alucio/components/Virtual/Menu/Menu.tsx [hunk 8]
```diff
@@ -214,28 +188,31 @@ function Menu(props: MenuProps) {
...
+                  setSettingsToggle((p) => !p);
                   props.onSelectedOption(...args)
                 }}
               />
             </Popover.Anchor>
             <Popover.Content offset={12}>
-              { menuOption.settingsParams
-                ? <SettingsMenu settingsParams = {menuOption.settingsParams} onCloseSettings={() => setSettingsToggle(false)} /> : <></> }
+              {menuOption.settingsParams
+                ? <SettingsMenu
```
Comment (chibchombiano26) [inline alucio/components/Virtual/Menu/Menu.tsx:213]: iffy?

**Comment by alucio-echen** [comment]
Here's a screenshot with the latest updates:
image

**Review by chibchombiano26 — APPROVED** [review]
LGTM

**Pull request merged**